CHANGELOG.md

```diff
@@ -1,5 +1,7 @@
 ## master / unreleased
  - [FEATURE] Provide option to compress WAL records using Snappy. [#609](https://github.com/prometheus/tsdb/pull/609)
+ - [BUGFIX] Re-calculate block size when calling `block.Delete`.
+ - [CHANGE] `BlockStats` no longer holds size information and this is kept in memory.
```

Comment:
```suggestion
 - [CHANGE] `BlockStats` no longer holds (not precise) size information. This is now dynamically calculate while opening a block in memory.
```

Reply:
the size is recalculated NOT just when opening a block, but also when calling block.Delete so I prefer to keep this a bit more generic.

Reply:
Fair. Can we have then first fix `(not precise)` and then `This is now dynamically calculated and kept in memory.`

Reply:
what do you mean by `(not precise)'? If that is the missing meta size that it can be something like:
```
- [CHANGE] `BlockStats` no longer holds size information. This is now dynamically calculated and kept in memory. It also includes the meta file size which was not included before.
```

Reply:
perfect!